packages/camera/ios/Classes/CameraPlugin.m

```diff
@@ -22,179 +22,343 @@ @interface FLTSavePhotoDelegate : NSObject<AVCapturePhotoCaptureDelegate>
-  output.videoSettings =
-      @{(NSString *)kCVPixelBufferPixelFormatTypeKey : @(kCVPixelFormatType_32BGRA) };
-  [output setAlwaysDiscardsLateVideoFrames:YES];
-  [output setSampleBufferDelegate:self queue:dispatch_get_main_queue()];
-
-  AVCaptureConnection *connection =
-      [AVCaptureConnection connectionWithInputPorts:input.ports output:output];
-  if ([_captureDevice position] == AVCaptureDevicePositionFront) {
-    connection.videoMirrored = YES;
-  }
-  connection.videoOrientation = AVCaptureVideoOrientationPortrait;
-  [_captureSession addInputWithNoConnections:input];
-  [_captureSession addOutputWithNoConnections:output];
-  [_captureSession addConnection:connection];
-  _capturePhotoOutput = [AVCapturePhotoOutput new];
-  [_captureSession addOutput:_capturePhotoOutput];
-  return self;
+    self = [super init];
+    NSAssert(self, @"super init cannot be nil");
+    _captureSession = [[AVCaptureSession alloc] init];
+    AVCaptureSessionPreset preset;
+    if ([resolutionPreset isEqualToString:@"high"]) {
+        preset = AVCaptureSessionPresetHigh;
+    } else if ([resolutionPreset isEqualToString:@"medium"]) {
+        preset = AVCaptureSessionPresetMedium;
+    } else {
+        NSAssert([resolutionPreset isEqualToString:@"low"], @"Unknown resolution preset %@",
+                 resolutionPreset);
+        preset = AVCaptureSessionPresetLow;
+    }
+    _captureSession.sessionPreset = preset;
+    _captureDevice = [AVCaptureDevice deviceWithUniqueID:cameraName];
+    NSError *localError = nil;
+    _captureVideoInput =
+    [AVCaptureDeviceInput deviceInputWithDevice:_captureDevice error:&localError];
+    if (localError) {
+        *error = localError;
+        return nil;
+    }
+    CMVideoDimensions dimensions =
+    CMVideoFormatDescriptionGetDimensions([[_captureDevice activeFormat] formatDescription]);
+    _previewSize = CGSizeMake(dimensions.width, dimensions.height);
+
+    _captureVideoOutput = [AVCaptureVideoDataOutput new];
+    _captureVideoOutput.videoSettings =
+    @{(NSString *)kCVPixelBufferPixelFormatTypeKey : @(kCVPixelFormatType_32BGRA) };
+    [_captureVideoOutput setAlwaysDiscardsLateVideoFrames:YES];
+    [_captureVideoOutput setSampleBufferDelegate:self queue:dispatch_get_main_queue()];
+
+    AVCaptureConnection *connection =
+    [AVCaptureConnection connectionWithInputPorts:_captureVideoInput.ports output:_captureVideoOutput];
+    if ([_captureDevice position] == AVCaptureDevicePositionFront) {
+        connection.videoMirrored = YES;
+    }
+    connection.videoOrientation = AVCaptureVideoOrientationPortrait;
+    [_captureSession addInputWithNoConnections:_captureVideoInput];
+    [_captureSession addOutputWithNoConnections:_captureVideoOutput];
+    [_captureSession addConnection:connection];
+    _capturePhotoOutput = [AVCapturePhotoOutput new];
+    [_captureSession addOutput:_capturePhotoOutput];
+    [self setUpCaptureSessionForAudio];
```

Comment:
@luigi-agosti @holykou here is where I added the change to record audio with reference to the plist change.

https://developer.apple.com/documentation/avfoundation/avassetwriter/1387863-canaddinput

for the asset writer may help to solve this.